### PR TITLE
frontend: improve webpack config.

### PR DESCRIPTION
**Pull request by kraftwerk28**

### CODE REVIEW

File: packages/frontend/package.json [hunk 1]
```diff
@@ -12,12 +18,10 @@
...
     "react-hot-loader": "^4.7.1",
     "webpack": "^4.29.5",
     "webpack-cli": "^3.2.3",
     "webpack-dev-server": "^3.2.1"
-  },
-  "scripts": {
-    "start": "webpack-dev-server --config ./webpack.config.js --mode development"
   }
-}
+}
```
Comment (o-rumiantsev) [inline packages/frontend/package.json]: Add new line

File: packages/frontend/src/index-template.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<!DOCTYPE html>
```
Comment (o-rumiantsev) [inline packages/frontend/src/index-template.html:1]: Maybe place it to `public/index.html`?
@Kowalski0805 wdyt?

File: packages/frontend/webpack.config.js [hunk 3]
```diff
@@ -1,30 +1,53 @@
...
-  resolve: {
-    extensions: ['*', '.js', '.jsx'],
-  },
-  output: {
-    path: __dirname + '/dist',
-    publicPath: '/',
-    filename: 'bundle.js',
-  },
-  plugins: [
-    new webpack.HotModuleReplacementPlugin(),
```
Comment (o-rumiantsev) [inline packages/frontend/webpack.config.js:23]: why did you remove this?

File: packages/frontend/yarn-error.log [hunk 4]
```diff
@@ -0,0 +1,5093 @@
+Arguments:
```
Comment (o-rumiantsev) [inline packages/frontend/yarn-error.log]: Remove this file

File: packages/frontend/src/index.js [hunk 5]
```diff
@@ -1,9 +1,17 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
+import { hot } from 'react-hot-loader';
 
-ReactDOM.render(
-  <h1>Project template</h1>,
-  document.getElementById('app')
+import metarhia from './assets/metarhia.png';
```
Comment (o-rumiantsev) [inline packages/frontend/src/index.js]: This is just template project, so i think we do not need it here

File: packages/frontend/dist/index.html [hunk 6]
```diff
@@ -1,10 +1 @@
...
-<html>
-<head>
-    <title>Project template</title>
-</head>
-<body>
-    <div id="app"></div>
-    <script src="./bundle.js"></script>
-</body>
-</html>
+<!DOCTYPE html><html lang="en"><head><meta charset="UTF-8"><meta name="viewport" content="width=device-width,initial-scale=1"><meta http-equiv="X-UA-Compatible" content="ie=edge"><title>Project template</title></head><body><div id="app"></div><script type="text/javascript" src="/bundle.js"></script></body></html>
```
Comment (o-rumiantsev) [inline packages/frontend/dist/index.html]: Do we need this file here, or it will be generated by `yarn build`?

File: package.json [hunk 7]
```diff
@@ -6,6 +6,8 @@
   "author": "Rumiantsev Oleksii <rumiantsev.oleksii@gmail.com>",
   "license": "MIT",
   "devDependencies": {
-    "lerna": "^3.13.0"
+    "file-loader": "^3.0.1",
+    "lerna": "^3.13.0",
+    "mini-css-extract-plugin": "^0.5.0"
```
Comment (o-rumiantsev) [inline package.json]: I think it is not needed

**Comment by o-rumiantsev** [comment]
Rename PR and commit title to `frontend: improve webpack config` or something like this

File: packages/frontend/src/index-template.html (hunk 2 above)
Comment (pogo5422) [reply packages/frontend/src/index-template.html:1]: This file is used by bundler (webpack plugin) so it's part of sources.

File: packages/frontend/webpack.config.js (hunk 3 above)
Comment (pogo5422) [reply packages/frontend/webpack.config.js:23]: It's more efficient way, when using `--hot` flag instead of placing this option inside config. So we don't need HMR plugin and `hot: true` in config

File: packages/frontend/dist/index.html (hunk 6 above)
Comment (pogo5422) [reply packages/frontend/dist/index.html]: Yes, it'll be generated by webpack. Gonna remove it from git track.

File: packages/frontend/src/index-template.html (hunk 2 above)
Comment (o-rumiantsev) [reply packages/frontend/src/index-template.html:1]: We can change this option in config i.e. `template: ./src/public/index.html`

Comment (Kowalski0805) [reply packages/frontend/src/index-template.html:1]: Create-react app usually places index.html in src/public, but it's up to you where to place this file.

**Review by Kowalski0805 — APPROVED** [review]

File: packages/frontend/src/index.js [hunk 8]
```diff
@@ -1,9 +1,12 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
+import { hot } from 'react-hot-loader';
 
-ReactDOM.render(
-  <h1>Project template</h1>,
-  document.getElementById('app')
+const App = () => hot(module)(
```
Comment (o-rumiantsev) [inline packages/frontend/src/index.js]: What this does and can we get around writing this. How do we connect non-functional components with hot loader?

**Review by o-rumiantsev — APPROVED** [review]

**Comment by o-rumiantsev** [comment]
Landed in 7fd5dae90cc76209ecdec2856fd5182c5ad6c563
